src/Microsoft.AspNetCore.Razor.Design/build/netstandard2.0/Microsoft.AspNetCore.Razor.Design.targets

```diff
@@ -42,45 +41,71 @@
   </PropertyGroup>
 
   <!-- 
-    Default values for properties that affect Razor MSBuild behavior. 
+    Default values for properties that affect Razor targets to the standard build lifecycle.
   -->
-  <PropertyGroup>
-    <!--
-      Razor also attaches itself by default to some of the standard .NET targets. Uses these properties to
-      configure this behaviour.
-    -->
-    <RazorCompileOnBuild Condition="'$(RazorCompileOnBuild)'==''">false</RazorCompileOnBuild>
+  <PropertyGroup Condition="'$(RazorCompileOnBuild)'==''">
+    <RazorCompileOnBuild>false</RazorCompileOnBuild>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(RazorCompileOnPublish)'==''">
+    <!-- Always compile on publish by default if we're compiling on build -->
+    <RazorCompileOnPublish Condition="'$(RazorCompileOnBuild)'=='true'"/>
+
+    <!-- Compatibility with the old MVC Precompilation setting -->
+    <RazorCompileOnPublish Condition="'$(RazorCompileOnPublish)'==''">$(MvcRazorCompileOnPublish)</RazorCompileOnPublish>
```

Comment:
The ` Condition="'$(RazorCompileOnPublish)'==''"` is guarunteed to be `true` since it's on the property group

Reply:
It gets re-evaluated on live 52.